src/c-opentimelineio/opentime-bindings/src/timeRange.cpp

```diff
@@ -0,0 +1,130 @@
+#include "copentime/timeRange.h"
+#include <opentime/rationalTime.h>
+#include <opentime/timeRange.h>
+
+#ifdef __cplusplus
+extern "C"
+{
+#endif
+    TimeRange* TimeRange_create()
+    {
+        return reinterpret_cast<TimeRange*>(new opentime::TimeRange());
+    }
+    TimeRange* TimeRange_create_with_start_time(RationalTime* start_time)
+    {
+        return reinterpret_cast<TimeRange*>(new opentime::TimeRange(
+            *reinterpret_cast<opentime::RationalTime*>(start_time)));
+    }
+    TimeRange* TimeRange_create_with_start_time_and_duration(
+        RationalTime* start_time, RationalTime* duration)
+    {
+        return reinterpret_cast<TimeRange*>(new opentime::TimeRange(
+            *reinterpret_cast<opentime::RationalTime*>(start_time),
+            *reinterpret_cast<opentime::RationalTime*>(duration)));
+    }
+    RationalTime* TimeRange_start_time(TimeRange* self)
```

Comment:
For efficiency, we could also include functions like this, where there is no new allocation, as long as the caller passes in a pre-allocated RationalTime.

ErrorStatus TimeRange_copy_start_time(TimeRange* self, RationalTime* output)
{
    *output = self->start_time();
    return SUCCESS;
}

The error checking part is a separate issue to the memory allocation part...

Reply:
This is for another day, so resolving the conversation.